### PR TITLE
Combined implementation of near and myriad API authentication

### PR DESCRIPTION
**Pull request by ical10**
# Title
- [x] JIRA Issue ID (preferably only ONE parent issue, put subtask IDs in the Description)
- [ ] Multiple JIRA Issue IDs? Are you sure your PR isn't too big? Can you rebase your commits to new branches so there's only 1 issue ID?
- [x] Is your title relatable to the JIRA issue?
- [x] Title <= 100 characters

# Description
Implemented authentication on Myriad API via NEAR wallet

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Builds and runs on Chrome Desktop
- [x] Builds and runs on Chrome Responsive
- [ ] Unit tests?
- [ ] Integration tests?
- [ ] End-to-end (E2E) tests?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Have you added yourself as the Assignee?
- [x] Have you added 1 or more leads as the Reviewer?
- [x] Have you chosen 1-2 of the following labels?
  - Feature
  - Bug
  - Refactor
  - Test
  - Documentation

### CODE REVIEW

**Comment by mucktar13** [comment]
@ical10 ignore .env.local bro

File: src/components/Login/render/Options/Options.tsx [hunk 1]
```diff
@@ -28,12 +28,16 @@ import {
...
 import {usePolkadotExtension} from 'src/hooks/use-polkadot-app.hook';
+import {WalletTypeEnum} from 'src/lib/api/ext-auth';
 import i18n from 'src/locale';
 
 type OptionProps = {
   network?: string;
-  onConnect: (accounts: InjectedAccountWithMeta[]) => void;
+  onConnect?: (accounts: InjectedAccountWithMeta[]) => void;
+  onConnectNear?: (nearId: string, callback: () => void) => void;
 };
```
Comment (mucktar13) [inline src/components/Login/render/Options/Options.tsx:41]: mending bikin `onConnect?: (wallet: WalletType, options: {accounts?: InjectedAccountWithMeta[], accountId?: string} ) => void`  aja.
kalo kayak gini makin banyak wallet makin banyak props

File: .env-template [hunk 2]
```diff
@@ -31,3 +31,16 @@ SENTRY_AUTH_TOKEN=
...
+
+# NEAR
+# TESTNET
+NETWORK_ID=
+NODE_URL=
+WALLET_URL=
+HELPER_URL=
+
+# MAINNET
+#NETWORK_ID=
```
Comment (irmannmal) [inline .env-template:43]: please remove this

File: .env-template [hunk 3]
```diff
@@ -31,3 +31,16 @@ SENTRY_AUTH_TOKEN=
 
 # FACEBOOK
 NEXT_PUBLIC_FACEBOOK_APP_ID=
+
+# NEAR
+# TESTNET
+NETWORK_ID=
```
Comment (irmannmal) [inline .env-template:37]: i'm suggest to use prefix identifier, e.g `NEAR_NETWORK_ID`

BTW can we use this configuration from database ? @ical10 @abdulhakim2902

**Review by irmannmal — CHANGES_REQUESTED** [review]

File: src/components/Login/render/Options/Options.tsx [hunk 4]
```diff
@@ -143,10 +173,19 @@ export const Options: React.FC<OptionProps> = props => {
             selected={network === NetworkTypeEnum.POLKADOT}
             onClick={setSelectedNetwork(NetworkTypeEnum.POLKADOT)}>
             <div className={styles.card}>
-              <PolkadotNetworkIcon className={styles.icon} />
```
Comment (mucktar13) [inline src/components/Login/render/Options/Options.tsx:146]: note aja bor, kalo ga dipake mending hapus aja icon nya dan reuse name yang sama

**Comment by ical10** [comment]
feedback done @mucktar13 @irmannmal 
can we merge this?

File: next.config.js [hunk 5]
```diff
@@ -43,6 +43,10 @@ const moduleExports = {
     firebaseAppId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
     firebaseMeasurementId: process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID,
     facebookAppId: process.env.NEXT_PUBLIC_FACEBOOK_APP_ID,
+    nearNetworkId: process.env.NEAR_NETWORK_ID ?? 'testnet',
```
Comment (irmannmal) [inline next.config.js:46]: is this used @ical10 ? look likes `NEAR_NETWORK_ID` deleted from .env-template

Comment (ical10) [reply next.config.js:46]: readded these fields back into `.env-template`, yes these are necessary env vars

Comment (irmannmal) [reply next.config.js:46]: got it!

**Review by irmannmal — APPROVED** [review]

**Review by kriptonhaz — APPROVED** [review]

**Pull request merged**